modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IORace.kt

```diff
@@ -0,0 +1,480 @@
+                  r3.fold({ e3 -> Platform.composeErrors(error, e3) }, { error })
+                })
+                cb(Left(errorResult))
+              }
+            }
+          } else {
+            promiseB.complete(Left(error))
+          }
+        }, { b ->
+          if (active.getAndSet(false)) {
+            conn.pop()
+            cb(Right(RaceTriple.Second(IOFiber(promiseA, connA), b, IOFiber(promiseC, connC))))
+          } else {
+            promiseB.complete(Right(b))
+          }
+        })
+      }
+
+      IORunLoop.startCancelable(IOForkedStart(ioC, ctx), connC) { either: Either<Throwable, C> ->
+        either.fold({ error ->
+          if (active.getAndSet(false)) { // if an error finishes first, stop the race.
+            connA.cancel().fix().unsafeRunAsync { r2 ->
+              connB.cancel().fix().unsafeRunAsync { r3 ->
+                conn.pop()
+                val errorResult = r2.fold({ e2 ->
+                  r3.fold({ e3 -> Platform.composeErrors(error, e2, e3) }, { Platform.composeErrors(error, e2) })
+                }, {
+                  r3.fold({ e3 -> Platform.composeErrors(error, e3) }, { error })
+                })
+                cb(Left(errorResult))
+              }
+            }
+          } else {
+            promiseC.complete(Left(error))
+          }
+        }, { c ->
+          if (active.getAndSet(false)) {
+            conn.pop()
+            cb(Right(RaceTriple.Third(IOFiber(promiseA, connA), IOFiber(promiseB, connB), c)))
+          } else {
+            promiseC.complete(Right(c))
+          }
+        })
+      }
+    }
+
+  /**
+   * Race two tasks concurrently within a new [F] on [this@raceN].
+   * At the end of the race it automatically cancels the loser.
+   *
+   * ```kotlin:ank:playground
+   * import arrow.Kind
+   * import arrow.fx.*
+   * import arrow.fx.typeclasses.Concurrent
+   * import kotlinx.coroutines.Dispatchers
+   * import arrow.fx.extensions.io.concurrent.concurrent
+   *
+   * fun main(args: Array<String>) {
+   *   fun <F> Concurrent<F>.example(): Kind<F, String> {
+   *     val never: Kind<F, Int> = cancelable { effect { println("Never got canelled for losing.") } }
+   *
+   *     //sampleStart
+   *     val result = fx.concurrent {
+   *       val eitherGetOrUnit = !Dispatchers.Default.raceN(never, just(5))
+   *       eitherGetOrUnit.fold(
+   *         { "Never always loses race" },
+   *         { i -> "Race was won with $i" }
+   *       )
+   *     }
+   *     //sampleEnd
+   *     return result
+   *   }
+   *
+   *   IO.concurrent().example().fix().unsafeRunSync().let(::println)
+   * }
+   * ```
+   *
+   * @param this@raceN [CoroutineContext] to execute the source [F] on.
+   * @param fa task to participate in the race
+   * @param fb task to participate in the race
+   * @return [F] either [Left] if [fa] won the race,
+   *   or [Right] if [fb] won the race.
+   *
+   * @see racePair for a version that does not automatically cancel the loser.
```

Comment:
I guess this doesn't apply anymore. BTW @nomisRev i guess this was referred to parMapN or something else?